mesonbuild/interpreter/interpreter.py

```diff
@@ -1617,6 +1617,7 @@ def notfound_program(self, args: T.List[mesonlib.FileOrString]) -> ExternalProgr
     # the host machine.
     def find_program_impl(self, args: T.List[mesonlib.FileOrString],
                           for_machine: MachineChoice = MachineChoice.HOST,
+                          default_options: T.Dict[OptionKey, T.Union[str, int, bool, T.List[str]]] | None = None,
```

Comment:
We usually use T.Union[foo, bar] instead of `foo | bar`, but I don't know why. Maybe for compat with older python version? @dcbaker @eli-schwartz ^

Reply:
In this case it's rather `T.Optional[T.Dict[...]]`

Reply:
Yes, that is the case. https://peps.python.org/pep-0604/

Reply:
Fixed